manywheel/build_docker.sh

```diff
@@ -57,9 +57,10 @@ DOCKER_IMAGE=${DOCKER_REGISTRY}/pytorch/manylinux-builder:${DOCKER_TAG}
     docker tag ${DOCKER_IMAGE} ${LEGACY_DOCKER_IMAGE}
 )
 
-if [[ -n ${WITH_PUSH} ]]; then
+if [[ "${WITH_PUSH}" == true ]]; then
     (
         set -x
+        echo "${DOCKER_TOKEN}" | docker login -u "${DOCKER_ID}" --password--stdin
```

Comment:
Let's try and have the login just be a separate step in the workflow, we don't want to have users to have to provide these environment variables in order to actually run this script